packages/gatsby-remark-images-contentful/src/utils/index.js

```diff
@@ -1,20 +1,34 @@
+const { URL } = require(`url`)
 const axios = require(`axios`)
 
-const getBase64Img = async url => {
-  const response = await axios({
-    method: `GET`,
-    responseType: `arraybuffer`,
-    url: `${url}`,
-  })
+// This should be replaced with the solution for https://github.com/gatsbyjs/gatsby/issues/24220
+const getBase64Img = async (url, reporter) => {
+  try {
+    const response = await axios({
+      method: `GET`,
+      responseType: `arraybuffer`,
+      url: `${url}`,
+    })
 
-  const base64Img = `data:${
-    response.headers[`content-type`]
-  };base64,${new Buffer(response.data).toString(`base64`)}`
+    const base64Img = `data:${
+      response.headers[`content-type`]
+    };base64,${new Buffer(response.data).toString(`base64`)}`
 
-  return base64Img
+    return base64Img
+  } catch (err) {
+    reporter.panic(`Failed downloading the base64 image for ${url}`, err)
+    return undefined
+  }
 }
 
-const buildResponsiveSizes = async ({ metadata, imageUrl, options = {} }) => {
+const buildResponsiveSizes = async (
+  { metadata, imageUrl, options = {} },
+  reporter
+) => {
+  const imageURL = new URL(
+    imageUrl.indexOf(`/`) === 0 ? `https:${imageUrl}` : imageUrl
```

Comment:
While probably correct, is `https` intentional here? Room to break versus `http`? (Also consider local development)

Reply:
we default to https to ensure we have encryption. Mixing using https assets within http should be no issue while developing. Also, this was part of the code before.

Reply:
Still unclear to me whether this change will work in develop @ localhost, where it's `http`? (You can get `https` to work but that's a lot of effort) It seems this particular line is hardcoding https, that's why I'm asking.